src/language_features/inlay_hints.rs

```diff
@@ -0,0 +1,105 @@
+use lsp_types::{
+    request::{InlayHintRequest, Request},
+    InlayHint, InlayHintLabel, InlayHintParams, Position, Range, TextDocumentIdentifier, Url,
+};
+use serde::Deserialize;
+
+use crate::{
+    context::Context,
+    position::lsp_position_to_kakoune,
+    types::{EditorMeta, EditorParams},
+    util::{editor_quote, escape_tuple_element},
+};
+
+#[derive(Debug, PartialEq, Clone, Default, Serialize, Deserialize)]
+struct InlayHintsOptions {
```

Comment:
good point defining this here. I just realized that a lot of types in `src/types.rs` should rather be in the module where they're used.
The name collision with InlayHintsParams is ugly but fair enough.